builtin/credential/aws/path_role_test.go

```diff
@@ -986,6 +986,65 @@ func TestAwsVersion(t *testing.T) {
 	}
 }
 
+// This test was used to reproduce https://github.com/hashicorp/vault/issues/7418
+// and verify its fix.
+func TestRoleResolutionWithSTSEndpointConfigured(t *testing.T) {
+	t.Skip("skipping test because it hits real endpoints")
```

Comment:
Any reason not to just make this an acceptance test where hitting real endpoints happens all the time?

Reply:
I'm uncomfortable with that for a couple of reasons. It can make tests flaky if the target endpoint has a period of unavailability. Also, since tests can be run by individual developers and CI environments, I've seen situations where live tests caused rate-limiting to be applied to everyone sharing an account.

Reply:
I think I might have taken that to literally. :sweat_smile: That actually makes totally sense as a suggestion when it's behind a VAULT_ACC flag and skips the test if the variables it needs are unset. Added!